web/components/widgets/editor.tsx

```diff
@@ -189,40 +190,44 @@ export function TextEditor(props: {
   )
 }
 
-export function RichContent(props: {
-  content: JSONContent | string
+function RichContent(props: {
+  content: JSONContent
   className?: string
   size?: 'sm' | 'md' | 'lg'
 }) {
   const { className, content, size = 'md' } = props
-  const editor = useEditor({
-    editorProps: { attributes: { class: proseClass(size) } },
-    extensions: [
-      StarterKit,
-      size === 'md' ? DisplayImage : Image,
-      DisplayLink.configure({ openOnClick: false }), // stop link opening twice (browser still opens)
-      DisplayMention,
-      DisplayContractMention,
-      GridComponent,
-      StaticReactEmbedComponent,
-      Iframe,
-      TiptapTweet,
-      TiptapSpoiler.configure({
-        spoilerOpenClass: 'rounded-sm bg-gray-200 cursor-text',
-        spoilerCloseClass:
-          'rounded-sm bg-gray-600 text-transparent [&_*]:invisible cursor-pointer select-none',
-      }),
-    ],
-    content,
-    editable: false,
-  })
-  useEffect(
-    // Check isDestroyed here so hot reload works, see https://github.com/ueberdosis/tiptap/issues/1451#issuecomment-941988769
-    () => void !editor?.isDestroyed && editor?.commands?.setContent(content),
-    [editor, content]
+
+  const html = useMemo(
+    () =>
+      generateHTML(content, [
+        StarterKit,
+        size === 'sm' ? DisplayImage : Image,
+        DisplayLink.configure({ openOnClick: false }), // stop link opening twice (browser still opens)
+        DisplayMention,
+        DisplayContractMention,
+        GridComponent,
+        StaticReactEmbedComponent,
+        Iframe,
+        TiptapTweet,
+        TiptapSpoiler.configure({
+          spoilerOpenClass: 'rounded-sm bg-gray-200 cursor-text',
+          spoilerCloseClass:
+            'rounded-sm bg-gray-600 text-transparent [&_*]:invisible cursor-pointer select-none',
+        }),
+      ]),
+    [content, size]
   )
 
-  return <EditorContent className={className} editor={editor} />
+  // TODO: sanitize
```

Comment:
I know if you just paste in html tags into the editor it does sanitize stuff like that. But I think 3% that if API users add onClick attributes to the JSON it would actually get all the way through when rendered.